File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/IntConverter.java
package org.matilda.commands.types;

import com.google.protobuf.Any;
import com.google.protobuf.Int32Value;
import com.google.protobuf.InvalidProtocolBufferException;

public class IntConverter implements ProtobufConverter<Integer> {
    @Override
    public Int32Value convertToProtobuf(Integer object) {
        return Int32Value.newBuilder().setValue(object).build();
    }

    @Override
    public Integer convertFromProtobuf(Any object) throws InvalidProtocolBufferException {
        return object.unpack(Int32Value.class).getValue();
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/IntConverter.kt
package org.matilda.commands.types

import pbandk.unpack
import pbandk.wkt.Any
import pbandk.wkt.Int32Value

class IntConverter : ProtobufConverter<Int> {
    override fun convertToProtobuf(obj: Int): Int32Value {
        return Int32Value(obj)
    }

    override fun convertFromProtobuf(obj: Any): Int {
        return obj.unpack(Int32Value.Companion).value
    }
}
